coralproject/talk
# Dynamic Public Path

## What does this PR do?

Fixes https://github.com/coralproject/talk/issues/1241

Uses ensure dynamic imports use proper path/url via `__webpack_public_path__` as seen on https://webpack.js.org/configuration/output/#output-publicpath. It does so using the provided `STATIC_URL` that is provided from the server at runtime, rather than build time. This has the advantage of allowing reconfiguration of the static root without rebuilding the static assets.

## How do I test this PR?

1. Set the following in your environment: `TALK_ROOT_URL=http://localhost:8081/proxied`
2. Configure a webserver to proxy `http://localhost:8081/proxied/` to `http://localhost:3000/`
3. Build static assets + start server
4. Load http://localhost:8081/proxied/ as an admin
5. Visit the `Configure` tab, and check the box for "Ask Readers a Question"
6. Notice that the box loads correctly by requesting the following chunk: `http://localhost:8081/proxied/static/0.chunk.js`
